### PR TITLE
PHPStan (Level 3) on tests/

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | PHPStan (Level 3) on tests/
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Relative to #16471
| How to test?      | Github Actions are :green_circle: 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27125)

### CODE REVIEW

File: tests/Integration/Core/Stock/StockManagerTest.php [hunk 1]
```diff
@@ -388,9 +388,9 @@ public function __construct($stock_available, int $pack_stock_type = PackStockTy
         $this->stock_available = new FakeStockAvailable4759($stock_available);
     }
 
-    public function hasAttributes(): bool
+    public function hasAttributes(): int
```
Comment (PierreRambaud) [inline tests/Integration/Core/Stock/StockManagerTest.php]: Nope :thinking:

File: tests/Integration/PrestaShopBundle/Controller/Api/ApiTestCase.php [hunk 2]
```diff
@@ -134,7 +133,8 @@ private function mockContext(): Context
         $controllerMock = $this->mockController();
         $contextMock->controller = $controllerMock;
 
-        $contextMock->currency = (object) ['sign' => '$'];
+        $contextMock->currency = new \Currency();
```
Comment (PierreRambaud) [inline tests/Integration/PrestaShopBundle/Controller/Api/ApiTestCase.php]:
```suggestion
        $contextMock->currency = new Currency();
```

File: tests/Integration/Core/Stock/StockManagerTest.php (hunk 1 above)
Comment (Progi1984) [reply tests/Integration/Core/Stock/StockManagerTest.php]: Fixed

File: tests/Integration/PrestaShopBundle/Controller/Api/ApiTestCase.php (hunk 2 above)
Comment (Progi1984) [reply tests/Integration/PrestaShopBundle/Controller/Api/ApiTestCase.php]: Fixed

**Review by PierreRambaud — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thank you @Progi1984

**Pull request merged**